pyproject.toml

```diff
@@ -73,10 +73,9 @@ testpaths = 'tests'
 log_format = '%(name)s %(levelname)s: %(message)s'
 filterwarnings = [
     'error',
-    # Work around https://github.com/pytest-dev/pytest/issues/10977 for Python 3.12
-    'ignore:(ast\.Str|ast\.NameConstant|ast\.Num|Attribute s) is deprecated and will be removed.*:DeprecationWarning:',
-    # issue with pytz - https://github.com/stub42/pytz/issues/105 for Python 3.12
-    'ignore:datetime\.utcfromtimestamp\(\) is deprecated.*:DeprecationWarning:',
+    # Python 3.9 and below allowed truncation of float to integers in some
+    # cases, by not making this an error we can test for this behaviour
+    'ignore:(.+)Implicit conversion to integers using __int__ is deprecated',
```

Comment:
By adding this `ignore` it allows the deprecated behaviour on older versions, thus testing them. Without the Rust change in this PR I get test failures which are consistent with https://github.com/pydantic/pydantic/issues/9018.